File: simplex.java/src/main/java/kr/co/tmon/simplex/channels/Channel.java
package kr.co.tmon.simplex.channels;

import java.util.UUID;

public class Channel implements IChannel {

	private String id;
	
	@Override
	public String getId() {
		return id;
	}	
	
	public Channel() {
		id = UUID.randomUUID().toString();
	}
}

File: Simplex.kotlin/src/main/kotlin/kr/co/tmon/simplex/channels/Channel.kt
package kr.co.tmon.simplex.channels

import java.util.UUID

class Channel : IChannel{
    override val id: String = UUID.randomUUID().toString()

    override fun equals(other: Any?): Boolean {
        return when (other) {
            is IChannel -> this.id == other.id
            else -> false
        }
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}

